pkg/operator/ceph/controller/mirror_peer.go

```diff
@@ -0,0 +1,267 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package controller provides Kubernetes controller/pod/container spec items used for many Ceph daemons
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	// #nosec G101 since this is not leaking any hardcoded credentials, it's just the prefix of the secret name
+	poolMirrorBoostrapPeerSecretName = "pool-peer-token"
+	// #nosec G101 since this is not leaking any hardcoded credentials, it's just the prefix of the secret name
+	fsMirrorBoostrapPeerSecretName = "fs-mirror-peer-token"
+	// RBDMirrorBootstrapPeerSecretName #nosec G101 since this is not leaking any hardcoded credentials, it's just the prefix of the secret name
+	RBDMirrorBootstrapPeerSecretName = "rbdMirrorBootstrapPeerSecretName"
+	// FSMirrorBootstrapPeerSecretName #nosec G101 since this is not leaking any hardcoded credentials, it's just the prefix of the secret name
+	FSMirrorBootstrapPeerSecretName = "fsMirrorBootstrapPeerSecretName"
+
+	// #nosec G101 since this is not leaking any hardcoded credentials, it's just the secret name
+	peerCephKeyringSecretName = "rbd-mirror-peer-keyring"
+	// peerCephConfigKey is config map key of the peer ceph.conf
+	peerCephConfigKey               = "peerCephConfig"
+	peerCephConfigFileConfigMapName = "rbd-mirror-peer-config"
+
+	// peerCephKeyringKey is the secret key of the peer keyring
+	peerCephKeyringKey = "peerCephKeyring"
+
+	RBDKeyringPeerTemplate = `
+[client.%s]
+	key = %s
+`
+	FSKeyringPeerTemplate = `
+[%s]
+	key = %s
+`
+	CephConfPeerTemplate = `
+[global]
+	fsid = %s
+	mon_host = %s
+`
+)
+
+func CreateBootstrapPeerSecret(ctx *clusterd.Context, clusterInfo *cephclient.ClusterInfo, object client.Object, namespacedName types.NamespacedName, scheme *runtime.Scheme) (reconcile.Result, error) {
+	context := context.TODO()
+	var err error
+	var ns, name, daemonType string
+	var boostrapToken []byte
+	switch objectType := object.(type) {
+	case *cephv1.CephBlockPool:
+		ns = objectType.Namespace
+		name = objectType.Name
```

Comment:
nit: You could factor out `ns` and `name` assignments before the switch statement if you just cast it first